dask/dataframe/io/parquet/core.py

```diff
@@ -1020,9 +1024,11 @@ def to_parquet(
             BlockIndex((df.npartitions,)),
             # Pass in the original metadata to avoid
             # metadata emulation in `map_partitions`.
+            # If we do not want a metadata_file we will
+            # return 0s.
             # This is necessary, because we are not
             # expecting a dataframe-like output.
-            meta=df._meta,
+            meta=df._meta if write_metadata_file else (None, "i8"),
```

Comment:
You can roll back this change.

This `meta=` argument is only meant to pacify (i.e. lie to) `map_partitions`. The value passed here just needs to be pandas-compatible, but is not actually **used** in any way after it's type is checked.

Reply:
Done :)